pkg/sql/schema_changer.go

```diff
@@ -287,6 +287,12 @@ func (sc *SchemaChanger) maybeAddDropRename(
 			return false, nil
 		}
 
+		// This can happen if a change other than the drop originally
+		// scheduled the changer for this table. If that's the case,
+		// we still need to wait for the deadline to expire.
+		if d := table.GCDeadline; d != 0 && timeutil.Since(timeutil.Unix(0, d)) < 0 {
+			return false, nil
+		}
```

Comment:
It's not clear why you need this check here. TRUNCATE will mark the descriptor as DROPPED and it should follow the normal asynchronous code path that deletes the data after the TTL.

Reply:
In the unittest that fails, two different schema change operations happen for the same table. The one to backfill a column gets scheduled and then this code path gets tripped, even though there's a GC deadline which has not yet expired. So we do need to check it here to ensure that we don't prematurely begin dropping.

Reply:
I guess we can't do much here. I'd have preferred returning an error here but returning nil allows the schema changer to be marked as complete and with the next gossip message received when the dropped descriptor appears another schema changer is scheduled with the execAfter time set correctly to the TTL deadline.